server/src/main/java/io/crate/planner/optimizer/joinorder/JoinGraph.java

```diff
@@ -247,7 +248,7 @@ private void addEdge(LogicalPlan from, Edge edge) {
                 if (values == null) {
                     values = Set.of(edge);
                 } else {
-                    values = new HashSet<>(values);
+                    values = new LinkedHashSet<>(values);
```

Comment:
is this the same for this usage?

Reply:
See https://github.com/crate/crate/pull/17100#discussion_r1932132977